ctxwaitgroup/ctxwaitgroup.go

```diff
@@ -0,0 +1,86 @@
+package ctxwaitgroup
+
+import (
+	"context"
+	"fmt"
+	"sync"
+)
+
+var ErrClosed = fmt.Errorf("waitgroup closed")
+var ErrNegWg = fmt.Errorf("neg waitgroup")
+
+type WaitGroup struct {
+	cond   *sync.Cond
+	closed bool
+
+	waiters int
+}
+
+func NewWaitGroup() *WaitGroup {
+	cond := sync.NewCond(&sync.Mutex{})
+	return &WaitGroup{
+		cond:   cond,
+		closed: false,
+	}
+}
+
+func (wg *WaitGroup) Inc() error {
```

Comment:
Rather than this returning an error, could we just return a bool for whether or not the operation was successful?
In the pubsub case, we're really just checking if we can acquire an addition to the wait group